lib/spf/util.rb

```diff
@@ -45,7 +45,7 @@ def self.ipv4_mapped_ipv6_address_pattern
   end
 
   def self.hostname
-    return @hostname ||= Socket.gethostbyname(Socket.gethostname).first
+    return @hostname ||= Addrinfo.getaddrinfo(Socket.gethostname, 0).first.getnameinfo.first
```

Comment:
I haven't audited this thoroughly to understand the implications, but I find this preserves existing behavior:
```suggestion
    return @hostname ||= Addrinfo.getaddrinfo(Socket.gethostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
```